### ISSUE TITLE
[supabase-js] Parser error for using aggregate and spread

### ISSUE THREAD

**Issue by thecoorum** [supabase/postgrest-js#524]
# Bug report

- [x] I confirm this is a bug with Supabase, not with my own application. 
- [x] I confirm I have searched the [Docs](https://docs.supabase.com), GitHub [Discussions](https://github.com/supabase/supabase/discussions), and [Discord](https://discord.supabase.com). 

## To Reproduce

Steps to reproduce the behavior, please provide code snippets or a repository:

1. Create a typed Supabase client with (types are generated with `npx supabase gen types typescript --local`
   ```
   createClient<Database>(...)
   ```
2. Try to query data
  2a. Using spread in `select`
        ```
        .select('...users (first_name, last_name)')
        ```
 2b. Using newly introduced aggregate functions
        ```
        .select('from:from_time.min(), till:till_time.max()')
        ```

2a will type fail with `PostgrestSingleResponse<ParserError<"Expected identifier at '...users (first_name, last_name'">>`
2b will type fail with `PostgrestSingleResponse<ParserError<"Unexpected input: .min(), till:till_time.max()">>`

## Expected behavior

No TypeScript errors

## System information

- OS: macOS
- Version of supabase-js: `^2.39.7`
- Version of Node.js: `19.4.0`

**Comment by stefan-girlich** [comment]
@encima Would you know if there is a workaround? Right now, I'm casting all outputs to `any` before further processing. Maybe there is an casting-based approach utilizing the Supabase types that are already generated.

**Comment by bnjmnt4n** [comment]
@stefan-girlich are you using the latest version of supabase-js?

**Comment by stefan-girlich** [comment]
Thanks for the hint. I upgraded supabase-js, but I'm now seeing a new error. I added an updated in my [original bug report](https://github.com/supabase/postgrest-js/issues/523#issuecomment-2047393866).

**Comment by avallete** [comment]
Hey everyone !

We've reworked the result inference in `postgrest-js` to address the typing issues mentioned above. The changes are available in a canary release, and we're actively seeking your feedback.

To test it out, update your `supabase-js` to version `2.46.0-rc.1` (which includes `postgrest-js` `1.17.0-rc.1`):

```bash
npm install supabase-js@2.46.0-rc.1
```

Please let us know if you encounter any issues, especially with retro-compatibility, so we can finalize it for the next release.

**Comment by avallete** [comment]
The fix has been released in `supabase-js` v2.46.0. I'm closing this issue, but feel free to reopen if you encounter any further errors.
